src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -305,7 +305,9 @@ public CSharpOperationFactory(SemanticModel semanticModel)
                 case BoundKind.TypeExpression:
                 case BoundKind.TypeOrValueExpression:
                 case BoundKind.UnconvertedCollectionExpression:
-
+                case BoundKind.UnconvertedConditionalOperator:
```

Comment:
>case BoundKind.UnconvertedConditionalOperator: [](http://example.com/codeflow?start=16&length=46)

Given other comments for this PR, I think this change is not expected. Besides, we should not be creating ```NoneOperation``` for language constructs that have IOperation representation (do not confuse with specific bound nodes). Effectively, ```NoneOperation``` represents a design debt for ```IOperation``` feature. I do not think the issue we are dealing with falls into the category. It looks like we are simply failing to properly handle an error scenario for a language construct that does have ```IOperation``` representation.